src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseGroupResource.java

```diff
@@ -154,8 +153,9 @@ public ExerciseGroupResource(StudentParticipationRepository studentParticipation
     @PreAuthorize("hasRole('EDITOR')")
     public ResponseEntity<ExerciseGroup> getExerciseGroup(@PathVariable Long courseId, @PathVariable Long examId, @PathVariable Long exerciseGroupId) {
         log.debug("REST request to get exercise group : {}", exerciseGroupId);
-        Optional<ResponseEntity<ExerciseGroup>> accessFailure = examAccessService.checkCourseAndExamAndExerciseGroupAccess(Role.EDITOR, courseId, examId, exerciseGroupId);
-        return accessFailure.orElseGet(() -> ResponseEntity.ok(exerciseGroupRepository.findByIdElseThrow(exerciseGroupId)));
+        ExerciseGroup exerciseGroup = exerciseGroupRepository.findByIdElseThrow(exerciseGroupId);
+        Optional<ResponseEntity<ExerciseGroup>> accessFailure = examAccessService.checkCourseAndExamAndExerciseGroupAccess(Role.EDITOR, courseId, examId, exerciseGroup);
```

Comment:
var might be more suitable than Optional<ResponseEntity<ExerciseGroup>> in this case

Reply:
So i tried this, the type then is "Optional<ResponseEntity<Object>>" instead of "Optional<ResponseEntity<X>>" and i would then have to cast it again or change the return type of the method, before i do that i rather keep it like this ^^